src/cdk-experimental/popover-edit/lens-directives.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ReplaySubject} from 'rxjs';
+import {Directive, ElementRef, EventEmitter, Input, OnDestroy, OnInit, Output} from '@angular/core';
+import {EDIT_PANE_SELECTOR} from './constants';
+import {closest} from './polyfill';
+import {EditRef} from './edit-ref';
+
+/**
+ * Options for what do to when the user clicks outside of an edit lens.
+ */
+export type PopoverEditClickOutBehavior = 'close' | 'submit' | 'nothing';
+
+/**
+ * A directive that attaches to a form within the edit lens.
+ * It coordinates the form state with the table-wide edit system and handles
+ * closing the edit lens when the form is submitted or the user clicks
+ * out.
+ */
+@Directive({
+  selector: 'form[cdkEditControl]',
+  host: {
+    '(ngSubmit)': 'onSubmit()',
+    '(keydown.enter)': 'onEnter(true)',
+    '(keyup.enter)': 'onEnter(false)',
+    '(keyup.escape)': 'onEscape()',
```

Comment:
We currently don't use the `keydown.xxx` syntax because, last time we tried, WebDriver didn't support setting the appropriate property on `KeyboardEvent`. I don't know if this is still true, though, since it was a rather long time ago.